docs/REST-API/18-PCL-Overview.md

```diff
@@ -0,0 +1,590 @@
+
+## Note About Comparing Integers and Floats
+
+[Note About Comparing Integers and Floats]:#note-about-comparing-integers-and-floats
+
+When comparing an integer with a float, the following rule applies: The number that has less precision is converted to the type of the other number before the comparison occurs.
+- If the floating point number is within the range `-2^53` to `2^53`, then the floating point has more precision than the integer, so the integer is converted to a float and the numbers are compared.
+- If the floating point number is outside that range, then the floating point representation cannot represent contiguous integers. So, the integer will have more precision, and the floating point number is converted to an integer and the numbers are compared.
+
+Since the floating point representation cannot represent all integers greater than `2^53`, i.e. `9007199254740992`, this leads to some awkward behavior. This is not considered a bug, but a recognized artifact of the inaccuracy of floating point in general and beyond this limit specifically. All programming languages that support floating point numbers have similar issues. Here are some examples with explanations:
+
+```
+iex> PCL.parse_and_evaluate("9007199254740992 == 9007199254740992.0", PCL.EvalContext.new())
+{:ok, {true, []}}
+```
+
+This is as expected the integer and the corresponding floating point number are equal.
+
+```
+iex> PCL.parse_and_evaluate("9007199254740992 == 9007199254740993.0", PCL.EvalContext.new())
+{:ok, {true, []}}
+```
+
+What is happening here is that `9007199254740993.0` cannot be represented in the floating point representation. So, the floating point number that is chosen is the closest one that _can_ be represented. That happens to be the original number `9007199254740992.0`. The equality then just compares the two numbers which are now the same.
+
+```
+iex> PCL.parse_and_evaluate("9007199254740992 == 9007199254740994.0", PCL.EvalContext.new())
+{:ok, {false, []}}
+```
+
+Again, as expected. `9007199254740994.0` is the next number that can be represented in floating point, so PCL can tell that they are not the same. Note that as the floats get larger, the gap between representable numbers also gets larger.
+
+For more information about floating point numbers and their peculiarities:
+
+- [What Every Programmer Should Know About Floating-Point Arithmetic](https://floating-point-gui.de)
+- [0.30000000000000004.com](https://0.30000000000000004.com)
+- [Floating Point Arithmetic: Issues and Limitations](https://docs.python.org/3/tutorial/floatingpoint.html)
+
+## Additional Functions
+
+Event Orchestration has added additional functions that can be used in PCL condtions. These functions support `threshold conditions`, which are condition that will evaluate to `true` when that particular condition is evaluated a certain number of times within a given time period. The two different threshold conditions are `trigger_count` and `resetting_trigger_count`.
```

Comment:
**Blocker:** I feel pretty strongly that `trigger_count over` and `resetting_trigger_count over` should be listed with the rest of the `Built-in Operators`. I believe the only reason they're listed separately right now is that they're implemented as a PCL custom function, which our end users don't need care about at all.

Reply:
Yes, this can be moved to the `Other Operations` section of `Built-in Operators`.

Reply:
I actually moved it to a new section before the `Other Operations` section.